server/src/main/java/org/elasticsearch/cluster/project/ProjectStateRegistry.java

```diff
@@ -95,12 +118,44 @@ public TransportVersion getMinimalSupportedVersion() {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeMap(projectsSettings);
+        if (out.getTransportVersion().onOrAfter(TransportVersions.PROJECT_STATE_REGISTRY_RECORDS_DELETIONS)) {
+            out.writeCollection(projectsMarkedForDeletion);
+            out.writeZLong(projectsMarkedForDeletionGeneration);
```

Comment:
Similarly for non-negative numbers
```suggestion
            out.writeVLong(projectsMarkedForDeletionGeneration);
```